Segfault when running example_sig.c

Hello,

I was trying the library for the first time and I'm getting a segmentation fault when trying to run [this example](https://github.com/open-quantum-safe/liboqs/wiki/Minimal-example-of-a-post-quantum-signature).

Here is the bt:
```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007ffff7bc3a58 in ?? () from /usr/lib/liboqs.so.2
#2  0x00007ffff7bc44c2 in ?? () from /usr/lib/liboqs.so.2
#3  0x00007ffff792c47e in pqcrystals_dilithium2_avx2_pointwise_acc_avx () from /usr/lib/liboqs.so.2
#4  0x00007ffff7931e93 in pqcrystals_dilithium2_avx2_nttunpack_avx () from /usr/lib/liboqs.so.2
#5  0x0000555555555280 in example_stack ()
#6  0x00005555555556f9 in main ()
```

I build liboqs 0.7.2 with these options (for more details check [this PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=liboqs) for Arch Linux):
```
cmake -G Ninja -B build -S "${pkgname}-${pkgver}"\
        -DBUILD_SHARED_LIBS='ON' \
        -DCMAKE_BUILD_TYPE='None' \
        -DCMAKE_INSTALL_PREFIX="${pkgdir}/usr" \
        -DOQS_BUILD_ONLY_LIB='OFF' \
        -DOQS_DIST_BUILD='ON' \
        -DOQS_USE_AES_OPENSSL='ON' \
        -DOQS_USE_SHA2_OPENSSL='ON' \
        -DOQS_USE_SHA3_OPENSSL='ON' \
        -DOQS_OPT_TARGET='generic' \
        -Wno-dev
```
And I'm using have glibc 2.36. Please let me know anything else I could do to help debug this issue. I will try to build liboqs with `CMAKE_BUILD_TYPE='Debug'`.

Actually, it's not just this example, but some other tests as well. I tried both with the release 0.7.2 and with the main branch and 19 tests fail for me.

<details>
  <summary>Logs</summary>

```
=================================== FAILURES ===================================
______________________________ test_kem[Kyber512] ______________________________
[gw5] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber512'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_kem(kem_name):
        if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
>       helpers.run_subprocess(
            [helpers.path_to_executable('test_kem'), kem_name],
        )

tests/test_cmdline.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_kem', 'Kyber512']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_kem Kyber512

__________________________ test_examples[example_sig] __________________________
[gw4] linux -- Python 3.10.9 /usr/bin/python3

program = 'example_sig'

    @helpers.filtered_test
    @pytest.mark.parametrize('program', ['example_kem', 'example_sig'])
    def test_examples(program):
>       helpers.run_subprocess(
            [helpers.path_to_executable(program)],
        )

tests/test_cmdline.py:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/example_sig'], working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/example_sig

______________________________ test_kem[Kyber768] ______________________________
[gw5] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber768'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_kem(kem_name):
        if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
>       helpers.run_subprocess(
            [helpers.path_to_executable('test_kem'), kem_name],
        )

tests/test_cmdline.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_kem', 'Kyber768']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_kem Kyber768

_____________________________ test_kem[Kyber1024] ______________________________
[gw5] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber1024'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_kem(kem_name):
        if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
>       helpers.run_subprocess(
            [helpers.path_to_executable('test_kem'), kem_name],
        )

tests/test_cmdline.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_kem', 'Kyber1024']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_kem Kyber1024

_____________________________ test_sig[Dilithium2] _____________________________
[gw5] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium2'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_sig(sig_name):
        if not(helpers.is_sig_enabled_by_name(sig_name)): pytest.skip('Not enabled')
        if sys.platform.startswith("win") and 'APPVEYOR' in os.environ:
            if 'SPHINCS' in sig_name and ('192f' in sig_name or '192s' in sig_name or '256f' in sig_name or '256s' in sig_name):
                pytest.skip('Skipping SPHINCS+ 192s and 256s tests on Windows AppVeyor builds')
>       helpers.run_subprocess(
            [helpers.path_to_executable('test_sig'), sig_name],
        )

tests/test_cmdline.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_sig', 'Dilithium2']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_sig Dilithium2

_____________________________ test_sig[Dilithium3] _____________________________
[gw5] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium3'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_sig(sig_name):
        if not(helpers.is_sig_enabled_by_name(sig_name)): pytest.skip('Not enabled')
        if sys.platform.startswith("win") and 'APPVEYOR' in os.environ:
            if 'SPHINCS' in sig_name and ('192f' in sig_name or '192s' in sig_name or '256f' in sig_name or '256s' in sig_name):
                pytest.skip('Skipping SPHINCS+ 192s and 256s tests on Windows AppVeyor builds')
>       helpers.run_subprocess(
            [helpers.path_to_executable('test_sig'), sig_name],
        )

tests/test_cmdline.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_sig', 'Dilithium3']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_sig Dilithium3

_____________________________ test_sig[Dilithium5] _____________________________
[gw5] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium5'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_sig(sig_name):
        if not(helpers.is_sig_enabled_by_name(sig_name)): pytest.skip('Not enabled')
        if sys.platform.startswith("win") and 'APPVEYOR' in os.environ:
            if 'SPHINCS' in sig_name and ('192f' in sig_name or '192s' in sig_name or '256f' in sig_name or '256s' in sig_name):
                pytest.skip('Skipping SPHINCS+ 192s and 256s tests on Windows AppVeyor builds')
>       helpers.run_subprocess(
            [helpers.path_to_executable('test_sig'), sig_name],
        )

tests/test_cmdline.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_sig', 'Dilithium5']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_sig Dilithium5

______________________________ test_kem[Kyber512] ______________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber512'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_kem(kem_name):
        kats = helpers.get_kats("kem")
        if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
>       output = helpers.run_subprocess(
            [helpers.path_to_executable('kat_kem'), kem_name],
        )

tests/test_kat.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/kat_kem', 'Kyber512']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/kat_kem Kyber512

______________________________ test_kem[Kyber768] ______________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber768'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_kem(kem_name):
        kats = helpers.get_kats("kem")
        if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
>       output = helpers.run_subprocess(
            [helpers.path_to_executable('kat_kem'), kem_name],
        )

tests/test_kat.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/kat_kem', 'Kyber768']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/kat_kem Kyber768

_____________________________ test_kem[Kyber1024] ______________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber1024'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_kem(kem_name):
        kats = helpers.get_kats("kem")
        if not(helpers.is_kem_enabled_by_name(kem_name)): pytest.skip('Not enabled')
>       output = helpers.run_subprocess(
            [helpers.path_to_executable('kat_kem'), kem_name],
        )

tests/test_kat.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/kat_kem', 'Kyber1024']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/kat_kem Kyber1024

_____________________________ test_sig[Dilithium2] _____________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium2'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_sig(sig_name):
        kats = helpers.get_kats("sig")
        if not(helpers.is_sig_enabled_by_name(sig_name)): pytest.skip('Not enabled')
>       output = helpers.run_subprocess(
            [helpers.path_to_executable('kat_sig'), sig_name],
        )

tests/test_kat.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/kat_sig', 'Dilithium2']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/kat_sig Dilithium2

_____________________________ test_sig[Dilithium3] _____________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium3'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_sig(sig_name):
        kats = helpers.get_kats("sig")
        if not(helpers.is_sig_enabled_by_name(sig_name)): pytest.skip('Not enabled')
>       output = helpers.run_subprocess(
            [helpers.path_to_executable('kat_sig'), sig_name],
        )

tests/test_kat.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/kat_sig', 'Dilithium3']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/kat_sig Dilithium3

_____________________________ test_sig[Dilithium5] _____________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium5'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_sig(sig_name):
        kats = helpers.get_kats("sig")
        if not(helpers.is_sig_enabled_by_name(sig_name)): pytest.skip('Not enabled')
>       output = helpers.run_subprocess(
            [helpers.path_to_executable('kat_sig'), sig_name],
        )

tests/test_kat.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/kat_sig', 'Dilithium5']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/kat_sig Dilithium5

____________________________ test_mem_kem[Kyber512] ____________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber512'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_mem_kem(kem_name):
        if not(helpers.is_kem_enabled_by_name(kem_name)):
            pytest.skip('Not enabled')
    
        Path(helpers.get_current_build_dir_name()+'/mem-benchmark').mkdir(parents=True, exist_ok=True)
    
        for i in range(3):
>          helpers.run_subprocess([helpers.path_to_executable('test_kem_mem'), kem_name, str(i)])

tests/test_mem.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_kem_mem', 'Kyber512', '0']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_kem_mem Kyber512 0

____________________________ test_mem_kem[Kyber768] ____________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber768'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_mem_kem(kem_name):
        if not(helpers.is_kem_enabled_by_name(kem_name)):
            pytest.skip('Not enabled')
    
        Path(helpers.get_current_build_dir_name()+'/mem-benchmark').mkdir(parents=True, exist_ok=True)
    
        for i in range(3):
>          helpers.run_subprocess([helpers.path_to_executable('test_kem_mem'), kem_name, str(i)])

tests/test_mem.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_kem_mem', 'Kyber768', '0']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_kem_mem Kyber768 0

___________________________ test_mem_kem[Kyber1024] ____________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

kem_name = 'Kyber1024'

    @helpers.filtered_test
    @pytest.mark.parametrize('kem_name', helpers.available_kems_by_name())
    def test_mem_kem(kem_name):
        if not(helpers.is_kem_enabled_by_name(kem_name)):
            pytest.skip('Not enabled')
    
        Path(helpers.get_current_build_dir_name()+'/mem-benchmark').mkdir(parents=True, exist_ok=True)
    
        for i in range(3):
>          helpers.run_subprocess([helpers.path_to_executable('test_kem_mem'), kem_name, str(i)])

tests/test_mem.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_kem_mem', 'Kyber1024', '0']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_kem_mem Kyber1024 0

___________________________ test_mem_sig[Dilithium2] ___________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium2'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_mem_sig(sig_name):
        if not(helpers.is_sig_enabled_by_name(sig_name)):
            pytest.skip('Not enabled')
    
        Path(helpers.get_current_build_dir_name()+'/mem-benchmark').mkdir(parents=True, exist_ok=True)
    
        for i in range(3):
>          helpers.run_subprocess([helpers.path_to_executable('test_sig_mem'), sig_name, str(i)])

tests/test_mem.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_sig_mem', 'Dilithium2', '0']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_sig_mem Dilithium2 0

___________________________ test_mem_sig[Dilithium3] ___________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium3'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_mem_sig(sig_name):
        if not(helpers.is_sig_enabled_by_name(sig_name)):
            pytest.skip('Not enabled')
    
        Path(helpers.get_current_build_dir_name()+'/mem-benchmark').mkdir(parents=True, exist_ok=True)
    
        for i in range(3):
>          helpers.run_subprocess([helpers.path_to_executable('test_sig_mem'), sig_name, str(i)])

tests/test_mem.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_sig_mem', 'Dilithium3', '0']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_sig_mem Dilithium3 0

___________________________ test_mem_sig[Dilithium5] ___________________________
[gw1] linux -- Python 3.10.9 /usr/bin/python3

sig_name = 'Dilithium5'

    @helpers.filtered_test
    @pytest.mark.parametrize('sig_name', helpers.available_sigs_by_name())
    def test_mem_sig(sig_name):
        if not(helpers.is_sig_enabled_by_name(sig_name)):
            pytest.skip('Not enabled')
    
        Path(helpers.get_current_build_dir_name()+'/mem-benchmark').mkdir(parents=True, exist_ok=True)
    
        for i in range(3):
>          helpers.run_subprocess([helpers.path_to_executable('test_sig_mem'), sig_name, str(i)])

tests/test_mem.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

command = ['/build/liboqs/src/build/tests/test_sig_mem', 'Dilithium5', '0']
working_dir = '.'
env = {'BUILDTOOL': 'devtools', 'BUILDTOOLVER': '20221012-2-any', 'CFLAGS': '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt...rror=format-security         -fstack-clash-protection -fcf-protection -flto=auto', 'CHOST': 'x86_64-pc-linux-gnu', ...}
expected_returncode = 0, input = None, ignore_returncode = False

    def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, input=None, ignore_returncode=False):
        """
        Helper function to run a shell command and report success/failure
        depending on the exit status of the shell command.
        """
        env_ = os.environ.copy()
        if env is not None:
            env_.update(env)
        env = env_
    
        # Note we need to capture stdout/stderr from the subprocess,
        # then print it, which pytest will then capture and
        # buffer appropriately
        print(working_dir + " > " + " ".join(command))
    
        result = subprocess.run(
                command,
                input=input,
                stdout=subprocess.PIPE,
                stderr=subprocess.STDOUT,
                cwd=working_dir,
                env=env,
            )
    
        if not(ignore_returncode) and (result.returncode != expected_returncode):
            print(result.stdout.decode('utf-8'))
>           assert False, "Got unexpected return code {}".format(result.returncode)
E           AssertionError: Got unexpected return code -11

tests/helpers.py:42: AssertionError
----------------------------- Captured stdout call -----------------------------
. > /build/liboqs/src/build/tests/test_sig_mem Dilithium5 0

=========================== short test summary info ============================
FAILED tests/test_cmdline.py::test_kem[Kyber512] - AssertionError: Got unexpe...
FAILED tests/test_cmdline.py::test_examples[example_sig] - AssertionError: Go...
FAILED tests/test_cmdline.py::test_kem[Kyber768] - AssertionError: Got unexpe...
FAILED tests/test_cmdline.py::test_kem[Kyber1024] - AssertionError: Got unexp...
FAILED tests/test_cmdline.py::test_sig[Dilithium2] - AssertionError: Got unex...
FAILED tests/test_cmdline.py::test_sig[Dilithium3] - AssertionError: Got unex...
FAILED tests/test_cmdline.py::test_sig[Dilithium5] - AssertionError: Got unex...
FAILED tests/test_kat.py::test_kem[Kyber512] - AssertionError: Got unexpected...
FAILED tests/test_kat.py::test_kem[Kyber768] - AssertionError: Got unexpected...
FAILED tests/test_kat.py::test_kem[Kyber1024] - AssertionError: Got unexpecte...
FAILED tests/test_kat.py::test_sig[Dilithium2] - AssertionError: Got unexpect...
FAILED tests/test_kat.py::test_sig[Dilithium3] - AssertionError: Got unexpect...
FAILED tests/test_kat.py::test_sig[Dilithium5] - AssertionError: Got unexpect...
FAILED tests/test_mem.py::test_mem_kem[Kyber512] - AssertionError: Got unexpe...
FAILED tests/test_mem.py::test_mem_kem[Kyber768] - AssertionError: Got unexpe...
FAILED tests/test_mem.py::test_mem_kem[Kyber1024] - AssertionError: Got unexp...
FAILED tests/test_mem.py::test_mem_sig[Dilithium2] - AssertionError: Got unex...
FAILED tests/test_mem.py::test_mem_sig[Dilithium3] - AssertionError: Got unex...
FAILED tests/test_mem.py::test_mem_sig[Dilithium5] - AssertionError: Got unex...
=========== 19 failed, 362 passed, 216 skipped in 127.25s (0:02:07) ============
FAILED: tests/CMakeFiles/run_tests /build/liboqs/src/build/tests/CMakeFiles/run_tests 
cd /build/liboqs/src/liboqs-0.7.2 && /usr/bin/cmake -E env OQS_BUILD_DIR=/build/liboqs/src/build python3 -m pytest --verbose --numprocesses=auto --ignore=scripts/copy_from_upstream/repos
ninja: build stopped: subcommand failed.
```
</details>

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -53,6 +53,10 @@ jobs:
             container: openquantumsafe/ci-alpine-amd64:latest
             CMAKE_ARGS: -DOQS_USE_OPENSSL=ON -DBUILD_SHARED_LIBS=ON
             PYTEST_ARGS: --ignore=tests/test_alg_info.py
+          - name: alpine-openssl-all
+            container: openquantumsafe/ci-alpine-amd64:latest
+            CMAKE_ARGS: -DOQS_USE_OPENSSL=ON -DBUILD_SHARED_LIBS=ON -DOQS_USE_AES_OPENSSL=ON -DOQS_USE_SHA2_OPENSSL=ON -DOQS_USE_SHA3_OPENSSL=ON
+            PYTEST_ARGS: --ignore=tests/test_alg_info.py
           - name: alpine-noopenssl
             container: openquantumsafe/ci-alpine-amd64:latest
             CMAKE_ARGS: -DOQS_USE_OPENSSL=OFF

--- a/src/common/sha3/ossl_sha3x4.c
+++ b/src/common/sha3/ossl_sha3x4.c
@@ -122,6 +122,10 @@ void OQS_SHA3_shake128_x4_inc_ctx_reset(OQS_SHA3_shake128_x4_inc_ctx *state) {
 	EVP_MD_CTX_reset(s->mdctx1);
 	EVP_MD_CTX_reset(s->mdctx2);
 	EVP_MD_CTX_reset(s->mdctx3);
+	EVP_DigestInit_ex(s->mdctx0, EVP_shake128(), NULL);
+	EVP_DigestInit_ex(s->mdctx1, EVP_shake128(), NULL);
+	EVP_DigestInit_ex(s->mdctx2, EVP_shake128(), NULL);
+	EVP_DigestInit_ex(s->mdctx3, EVP_shake128(), NULL);
 	s->n_out = 0;
 }
 
@@ -236,6 +240,10 @@ void OQS_SHA3_shake256_x4_inc_ctx_reset(OQS_SHA3_shake256_x4_inc_ctx *state) {
 	EVP_MD_CTX_reset(s->mdctx1);
 	EVP_MD_CTX_reset(s->mdctx2);
 	EVP_MD_CTX_reset(s->mdctx3);
+	EVP_DigestInit_ex(s->mdctx0, EVP_shake256(), NULL);
+	EVP_DigestInit_ex(s->mdctx1, EVP_shake256(), NULL);
+	EVP_DigestInit_ex(s->mdctx2, EVP_shake256(), NULL);
+	EVP_DigestInit_ex(s->mdctx3, EVP_shake256(), NULL);
 	s->n_out = 0;
 }
 

